src/renderers/native/ReactNativeFiber.js

```diff
@@ -0,0 +1,367 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactNativeFiber
+ * @flow
+ */
+
+'use strict';
+
+import type { Fiber } from 'ReactFiber';
+import type { ReactNodeList } from 'ReactTypes';
+
+const NativeMethodsMixin = require('NativeMethodsMixin');
+const ReactControlledComponent = require('ReactControlledComponent');
+const ReactFiberReconciler = require('ReactFiberReconciler');
+const ReactGenericBatching = require('ReactGenericBatching');
+const ReactInstanceMap = require('ReactInstanceMap');
+const ReactNativeAttributePayload = require('ReactNativeAttributePayload');
+const ReactNativeComponentTree = require('ReactNativeComponentTree');
+const ReactNativeInjection = require('ReactNativeInjection');
+const ReactNativeTagHandles = require('ReactNativeTagHandles');
+const ReactNativeViewConfigRegistry = require('ReactNativeViewConfigRegistry');
+const ReactPortal = require('ReactPortal');
+const UIManager = require('UIManager');
+
+const deepFreezeAndThrowOnMutationInDev = require('deepFreezeAndThrowOnMutationInDev');
+const findNodeHandle = require('findNodeHandle');
+const invariant = require('invariant');
+
+const { precacheFiberNode, uncacheFiberNode } = ReactNativeComponentTree;
+
+ReactNativeInjection.inject();
+// ReactControlledComponent.injection.injectFiberControlledHostComponent(
+//   ReactNativeFiberComponent
+// );
+
+type Container = mixed;
+type Props = mixed;
+type Instance = mixed;
+type TextInstance = mixed;
+
+function NativeHostComponent(tag) {
+  this._nativeTag = tag;
+  this._children = [];
+}
+Object.assign(NativeHostComponent.prototype, NativeMethodsMixin);
+
+function recursivelyUncacheFiberNode (node) {
+  if (typeof node === 'number') { // Leaf node (eg text)
+    uncacheFiberNode(node);
+  } else {
+    uncacheFiberNode(node._nativeTag);
+
+    node._children.forEach(recursivelyUncacheFiberNode);
+  }
+}
+
+const NativeRenderer = ReactFiberReconciler({
+
+  appendChild(
+    parentInstance : Instance | Container,
+    child : Instance | TextInstance
+  ) : void {
+    if (typeof parentInstance === 'number') {
+      // Root container
+      UIManager.setChildren(
+        parentInstance, // containerTag
+        [child._nativeTag] // reactTags
+      );
+    } else {
+      parentInstance._children.push(child);
+
+      UIManager.manageChildren(
+        parentInstance._nativeTag, // containerTag
+        [], // moveFromIndices
+        [], // moveToIndices
```

Comment:
If you are tired of forgetting the arguments I added flowtypings for this function in my PR https://github.com/facebook/react/pull/8361/files#diff-0480067610415f2abfec7a69ca45997eR37